### PR TITLE
Add in dynamic loss function support for verl Tinker

### PR DESCRIPTION
**Pull request by wyettzeng**
Make verl-Tinker more "Tinker" in nature. The client no longer have to pre-configure the loss functions in the configs, as the server now natively supports all 5 official tinker loss: `cross_entropy`, `importance_sampling`, `ppo`, `cispo`, and `dro`. However, for more advance users, they can also pass in `custom_from_config` and the server will use the config provided at startup.

Following the above change, the server also removes algorithm and data section from the config as those are not needed and dealt with on the client side.

### CODE REVIEW

File: verl_tinker/src/verl_tinker/data/datum_processing.py [hunk 1]
```diff
@@ -720,6 +729,14 @@ def _datums_to_forward_td(datums, pad_to_multiple: int = 1) -> TensorDict:
     for d in datums:
         prefix = list(d.model_input.to_ints())
         target_tokens = _tensor_data_to_torch(d.loss_fn_inputs["target_tokens"]).long()
+        if target_tokens.dim() != 1:
```
Comment (wyettzeng) [inline verl_tinker/src/verl_tinker/data/datum_processing.py:732]: For reviewer, in case you are wondering, we currently only support 1D forward. We are going to fail immediately at int(target_tokens[-1].item()), so might as well have this guard so that the error message is more clear. This limitation will most likely be addressed in another pr.

**Review by Luosuu — APPROVED** [review]
